src/containers/search/SearchForm.tsx

```diff
@@ -157,6 +166,28 @@ export const SearchForm: React.FC = () => {
     }
   }, [roomCount, adultCount, dateRange, location, refetch]);
 
+  // Prevent error messages from persisting on path change
+  // clear errors when path changes
+  const clearErrorMessages = useCallback(() => {
+    clearErrors();
+    setShowError(undefined);
+    setShowAccommodationsError(false);
+  }, []);
+
+  // set local error when error object changes
+  useEffect(() => {
+    setShowError(error);
+  }, [error, setShowError]);
```

Comment:
`setShowError` setter is a system function and React guarantees that this function is stable and will not be changed during a component lifecycle. So, you can exclude such callback from the dependencies of `useEffect`

Reply:
I will remove it. Thank you